### PR TITLE
Prevent bad practice in python tests

### PR DESCRIPTION
**Pull request by revans2**
In general trying to access the spark session outside of a `with_*_session` can result in tests that don't do what you want them to do. To help prevent this from happening this makes `spark` which was available for everyone to use before now `_spark` which shows that it is intended to be private.  It also adds a check when returning something from a `with_*_session` to try and avoid someone returning a dataframe.  When you create a dataframe with one config and then change configs spark has some lazy processing and you don't always get what you expect in terms of how that data was configured.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: integration_tests/src/main/python/spark_init_internal.py [hunk 1]
```diff
@@ -28,5 +28,12 @@ def _spark__init():
     _s.sparkContext.setLogLevel("WARN")
     return _s
 
-spark = _spark__init()
+_spark = _spark__init()
+
+def get_spark_i_know_what_i_am_doing():
+    """
+    Get the current SparkSession. Because of how tests work
```
Comment (tgravescs) [inline integration_tests/src/main/python/spark_init_internal.py:35]: it might be nice to state why

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@kuhushukla and @tgravescs could you take another look?

**Pull request merged**